### PR TITLE
Add the possibility to use pydantic date and datetime types

### PR DESCRIPTION
**Pull request by anneum**
Related to issue https://github.com/marcosschroh/dataclasses-avroschema/issues/510

### CODE REVIEW

**Review by marcosschroh — COMMENTED** [review]
I think for the new types could inherit from other fields to reuse the code. For example `PastDateField` could inherits from [DateField](https://github.com/marcosschroh/dataclasses-avroschema/blob/master/dataclasses_avroschema/fields/fields.py#L497) which contains the logic for `default_to_avro`.  The same for the new `datetime` fields

**Review by marcosschroh — COMMENTED** [review]
Can you add a new [test](https://github.com/marcosschroh/dataclasses-avroschema/blob/master/tests/model_generator/test_model_pydantic_generator.py) to check the `pydantic model generation`?

**Comment by anneum** [comment]
> I think for the new types could inherit from other fields to reuse the code. For example `PastDateField` could inherits from [DateField](https://github.com/marcosschroh/dataclasses-avroschema/blob/master/dataclasses_avroschema/fields/fields.py#L497) which contains the logic for `default_to_avro`. The same for the new `datetime` fields

I tested this and then had problems with the validation. Therefore the fields inherit from `PydanticField`, because `validate_field` (https://github.com/marcosschroh/dataclasses-avroschema/blob/master/dataclasses_avroschema/pydantic/fields.py#L14) is always true there. I don't know how we can combine the two requirements together.

**Comment by anneum** [comment]
> Can you add a new [test](https://github.com/marcosschroh/dataclasses-avroschema/blob/master/tests/model_generator/test_model_pydantic_generator.py) to check the `pydantic model generation`?

I added the new test.

File: dataclasses_avroschema/pydantic/fields.py [hunk 1]
```diff
@@ -210,6 +253,114 @@ def fake(self) -> uuid.UUID:
...
+            "pydantic-class": "PastDate",
+        }
+
+    def default_to_avro(self, date: PastDate) -> int:
+        return _default_to_avro_date(date)
+
+    def fake(self) -> datetime:
+        return fake.date_object()
+
+
```
Comment (marcosschroh) [inline dataclasses_avroschema/pydantic/fields.py:273]: Referencing to the inheritance issue that I pointed before. You can do this:

```python
class AwareDatetimeField(fields.DatetimeField):
    """Like datetime, with the constraint that the value must have timezone info."""

    @property
    def avro_type(self) -> typing.Dict:
        return {
            "type": LONG,
            "logicalType": TIMESTAMP_MILLIS,
            "pydantic-class": "AwareDatetime",
        }

    def validate_default(self, default) -> bool:
        msg = f"Invalid default type. Default should be {datetime}"
        assert isinstance(default, datetime), msg
        return True
```

We should do the same for all the fields and at some point get rid of the `PydanticField`

Comment (anneum) [reply dataclasses_avroschema/pydantic/fields.py:238]: I changed the base class as you suggested. For the other pydantic fields like `NegativeIntField `the problem is a little different. If I use the `fields.IntField` as superclass, the signature of "avro_type" is incompatible with supertype `fields.IntField`.

**Review by marcosschroh — APPROVED** [review]

**Pull request merged**